### PR TITLE
iOS: Added functionality to subscribe and unsubscribe to topics.

### PR DESCRIPTION
**Pull request by pablorueda**
## Description
Added functionality to subscribe and unsubscribe to topics.
This functionality is achieved implementing the optional methods to register to tags from our SDK.

## Checklist

* [x] PR is scoped to one task
* [ ] Tests are included
* [ ] Documentation is included

* [x] This PR is not making any UI change
* [ ] This PR is making a UI change
  * [ ] Screenshot(s) included

## QA :

* required test cases:
* specific apps / plugins to test :

### CODE REVIEW

**Review by rkarpievich-sc — CHANGES_REQUESTED** [review]
Add please parentheses to `error == nil`

**Comment by PRuedaApplicaster** [comment]
> Add please parentheses to `error == nil`

Done ;)

File: iOS/ZappPushPluginFirebase/APPushProviderFirebase.swift [hunk 1]
```diff
@@ -35,13 +35,56 @@ open class APPushProviderFirebase: ZPPushProvider {
...
+            completion(true, nil)
+            return
+        }
+        
+        var success = true
+        let dispatchGroup = DispatchGroup()
+        for tag in tags {
+            dispatchGroup.enter()
+            Messaging.messaging().subscribe(toTopic: tag) { (error) in
+                success = success && error == nil
```
Comment (rkarpievich-sc) [inline iOS/ZappPushPluginFirebase/APPushProviderFirebase.swift]: And here too

**Review by rkarpievich-sc — CHANGES_REQUESTED** [review]

File: iOS/ZappPushPluginFirebase/APPushProviderFirebase.swift (hunk 1 above)
Comment (PRuedaApplicaster) [reply iOS/ZappPushPluginFirebase/APPushProviderFirebase.swift]: Done :)

**Review by rkarpievich-sc — APPROVED** [review]

**Pull request merged**